### PR TITLE
Fr 117 s2s auth backend

### PR DESCRIPTION
**Pull request by padmajaruk**
### JIRA link ###
https://tools.hmcts.net/jira/browse/FR-117

### Change description ###
1.Explore how to connect to service2service auth connectivity
2. Create a method which accepts the authorisation header with valid Jwt token and parse the token successfully.
3. Create the variables and values for vault acces which are missing.
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: infrastructure/variables.tf [hunk 1]
```diff
@@ -44,3 +44,18 @@ variable "ilbIp"{}
...
+
+variable "auth_provider_service_client_baseurl" {
+  default = "http://betaDevBccidamS2SLB.reform.hmcts.net:80"
+}
+
+variable "auth_provider_service_client_tokentimetoliveinseconds" {
+  default = "900"
+
+  variable "vault_env" {}
+}
```
Comment (rafalkalita) [inline infrastructure/variables.tf]: One bracket too many

File: docker-compose.yml [hunk 2]
```diff
@@ -18,8 +18,26 @@ services:
...
+  service-auth-provider-api:
+      image: docker.artifactory.reform.hmcts.net/auth/service-auth-provider-api:latest
+      healthcheck:
+        disable: true
+      ports:
+        - 4502:8080
+      environment:
+        - 'SPRING_APPLICATION_JSON={
+            "auth.provider.service.testing-support.enabled" : true,
+            "auth.provider.service.server.jwtKey" : "wThK0f0/lh3FlxFcL4xUWDMI5C1J9KyQBgXV4wseh1e5J1uYJIjvTvArHxQDrYoHJ23xFxjHkOnvNbR5dXRoxA==",
```
Comment (rafalkalita) [inline docker-compose.yml]: Can we merge this key to the public repo? Is it only used in the local deployment?

File: infrastructure/variables.tf (hunk 1 above)
Comment (padmajaruk) [reply infrastructure/variables.tf]: done

File: docker-compose.yml (hunk 2 above)
Comment (padmajaruk) [reply docker-compose.yml]: This is what I found in divorce project, I can see this on confluence pages as well openly ( divorce project), I will once again confirm with Ganesh.

**Pull request merged**